tests/ui/almost_swapped.rs

```diff
@@ -0,0 +1,32 @@
+#![allow(clippy::needless_late_init, clippy::manual_swap)]
+#![allow(unused_variables, unused_assignments)]
+#![warn(clippy::almost_swapped)]
+
+fn main() {
+    let b = 1;
+    let a = b;
+    let b = a;
+
+    let mut c = 1;
+    let mut d = 2;
+    d = c;
+    c = d;
+
+    let mut b = 1;
+    let a = b;
+    b = a;
+
+    let b = 1;
+    let a = 2;
+
+    let t = b;
+    let b = a;
+    let a = t;
+
+    let mut b = 1;
+    let mut a = 2;
+
+    let t = b;
+    b = a;
+    a = t;
+}
```

Comment:
Let's stick these in the existing test file `swap.rs`, there might be some where the suggestions don't compile so you can add `mut` to make them compile or put them into a `swap_unfixable.rs`

Alternatively split out the existing tests from `swap.rs` into this `almost_swapped.rs` and create a `manual_swap.rs` for the rest

Reply:
There's still this one to cover ^

Reply:
@Alexendoo Sorry again. I moved these(L:26 ~ L:31) into `tests/ui/swap.rs` but I'm not certain this is correct.

Reply:
Ah I just clicked on the end of the file to comment, not meaning those couple lines specifically

I mean all the tests so there's one place `almost_swapped` is being tested, whether that's combining it into `swap.rs` or moving the existing `almost_swapped` tests out of `swap.rs` into `almost_swapped.rs` and renaming `swap.rs` to `manual_swap.rs`

Reply:
I appreciate your detailed exmplanation